derivation/src/main/scala/io/circe/magnolia/MagnoliaDecoder.scala

```diff
@@ -34,7 +34,11 @@ private[magnolia] object MagnoliaDecoder {
             val keyCursor = c.downField(key)
             keyCursor.focus match {
               case Some(json) => json.as[p.PType](p.typeclass)
-              case None => p.default.toRight(DecodingFailure("Attempt to decode value on failed cursor", keyCursor.history))
+              case None => p.default.map { default => Right(default) }.getOrElse {
```

Comment:
good, I would just use fold:
```
p.default.fold(p.typeclass.tryDecode(keyCursor))(Right(_))
```

Reply:
Sure! I've never liked Option.fold because it puts the `None` case first, which is not natural to me, but happy to make the change.